data/api/client-server/pushrules.yaml

```diff
@@ -365,9 +365,20 @@ paths:
     put:
       summary: Add or change a push rule.
       description: |-
-        This endpoint allows the creation, modification and deletion of pushers
-        for this user ID. The behaviour of this endpoint varies depending on the
-        values in the JSON body.
+        This endpoint allows the creation and modification of user defined push
+        rules.
+
+        If a rule with the same `rule_id` already exists among rules of the same
+        kind, it is updated with the new parameters, otherwise a new rule is
+        created.
+
+        If both `after` and `before` are provided, the new or updated rule must
+        be the next most important rule with respect to the rule identified by
+        `before`.
```

Comment:
In this case, Synapse ignore `after` completely, while Dendrite checks that the `after` rule exists and is before `before`.

Dendrite's behavior seems more sensible here, even if it seems there's no real value in setting them both anyway.